homeassistant/components/obihai/entity.py

```diff
@@ -0,0 +1,39 @@
+"""Obihai Entity base class."""
+from pyobihai import PyObihai
+
+from homeassistant.helpers.entity import DeviceInfo, Entity
+
+from .const import DOMAIN, OBIHAI
+
+
+class ObihaiEntity(Entity):
+    """Obihai Entity base Class."""
+
+    _attr_has_entity_name = True
```

Comment:
Looking at this part in the sensor platform:

![image](https://user-images.githubusercontent.com/195327/225304671-e358241b-4fa5-4853-9771-2e2cf153a8eb.png)

It seems like this integration isn't meeting the requirement for this flag. Entity names should start with a capital letter, the rest of the words are lower case (unless it's a proper noun or a capitalized abbreviation of course).

Reply:
Resolved, had to spin a new pyobihai wheel for this as we're using the service names directly from the Obihai REST API.  PyObihai now converts from title case to sentence case.

Reply:
Testing this I'm a bit concerned, the migration flow doesn't work well - The existing entities don't get used, they become "Unavailable" and duplicates all get created.  These new entity names aren't recognized as the same entities.  It's not a huge deal to delete and re-add the integration, but it's not seamlessly migrating.

It's possible some other change somewhere else introduced this, I'm not really sure how to resolve this at the moment.

Reply:
OK, I think I got this working with async_migrate_entries, since unique ID's are case sensitive, I'm opting to migrate everything to all-lower-case.  Latest PR shows this.